core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -330,6 +330,13 @@ object DataMirror {
     }.map(_._1).flatten
   }
 
+  /** Check if the given Data is fully aligned w.r.t base.  Chisel version of FIRRTL "passive".
+    *
+    * @param base Data to check full alignment for and relative to
+    * @return `true` if Data is fully aligned, `false` otherwise.
+    */
+  def isFullyAligned(base: Data): Boolean = collectFlippedDeep(base)(_ => false).isEmpty
```

Comment:
What about using `.containsAFlipped`? It's relatively new, might have been added around the time that you opened the original PR.

Reply:
Oh! Hadn't seen that.  Neat, seems useful to cache/avoid repeated computation of these recursive type properties like whether there's a flip.

Looks like that ignores direction coercion, this reports this information for the passed argument as it would be interpreted/used as-is.

Reply:
Hmm, I'll look at this more closely.  Thanks!

Reply:
Nevermind, it only works properly on HW things (not bare chisel types) and also reports if there are flip at all not if all members are aligned with the base, so it's similar/close but not what's being done here.

Thanks for pointing this out, though!